jax/_src/prng.py

```diff
@@ -578,6 +578,7 @@ def device_put_replicated(val, aval, sharding, devices):
 
 
 class KeyTy:
+  type = dtypes.prng_key
```

Comment:
No, `dtype` should not be a subtype of `dtypes.prng_key`. There are two type systems in numpy: scalar types (e.g. `np.float64`, which is a type that derives from `np.floating`, `np.numeric`, `np.generic`) and dtypes (`np.dtype('float64')`, which is an instance of type `np.dtype`). When you do `issubdtype(dtype, category)`, the `dtype` should generally be an instance of type `dtype`, and `category` should be something in the scalar type hierarchy.

With that in mind, I made `prng_key` derive from `np.generic`: it's the analog of `np.floating`. `KeyTy` is an analog of `np.dtype`, and just as `np.dtype` doesn't derive from `np.generic`, `KeyTy` should not derive from `prng_key`.

Reply:
On the other hand, we could decide to just break entirely from NumPy here... it might be less confusing.

Reply:
Thanks. I wasn't thinking clearly about the Numpy way of things. I think it makes sense not to break from it since we don't need to.

I'd still like to avoid having `KeyTy` be a public symbol. Maybe the other thread (re. common interface) can lead us to return something other than `KeyTy` from `keys.dtype`, at least when it comes to the type annotation of the `.dtype` attribute?

Reply:
Done in #16795